### PR TITLE
Improve WP CLI Extendability

### PR DESCRIPTION
**Pull request by crstauf**
PR for #265.

### CODE REVIEW

File: classes/ActionScheduler_Abstract_WPCLI_Command.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	/**
+	 * @var bool|string Enable timestamp, or timestamp format.
+	 */
+	protected $timestamp_string = false;
+
+	/**
+	 * Construct.
+	 */
+	function __construct( $args, $assoc_args ) {
```
[the remark this answers is not in the document]
Comment (crstauf) [reply classes/ActionScheduler_Abstract_WPCLI_Command.php]: @JPry I've made this change.

I'm wondering: if the function's scope defaults to `public`, why explicitly declare it?

**Comment by crstauf** [comment]
@JPry Submitted fixes and replies to your comments.

**Comment by crstauf** [comment]
@JPry I've also created https://github.com/crstauf/action-scheduler/pull/1 which adds `version` command to `issue_265`. 

I understand that the [second commit](https://github.com/crstauf/action-scheduler/pull/1/commits/5750802b0bfa4dca7ba4750551907af9ce60ba55) may not be wanted, as it makes some changes to `ActionScheduler_Versions` class; I went ahead and built it, as I see benefit, but also recognize it may not be wanted.

Screenshot of command output:

Screen Shot 2019-03-27 at 12 17 11 AM

File: classes/ActionScheduler_Abstract_WPCLI_Command.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (JPry) [reply classes/ActionScheduler_Abstract_WPCLI_Command.php]: There are 2 main reasons:

1. Being explicit ensures there is no confusion about intention. Leaving the scope undeclared allows for the question "was this intentional?". Being explicit removes this question
1. Consistency with other methods.

**Comment by JPry** [comment]
@crstauf Regarding the separate command additions, I think those would best be handled as separate PRs to this repo. I think version command is a good idea, but I'll have to look more closely at how it might impact the library. That's one where I would also want feedback from @thenbrent. The add command seems like a nice idea as well, but I'd like to give it a more thorough review here so it can have the attention it deserves.

Would you minding opening PRs for those here?

**Comment by crstauf** [comment]
> Would you mind opening PRs for those here?

If I do, then the PRs will be for everything, not just the commands themselves. Are you unable to leave a review on the existing PR?

If not, then I'd prefer for this PR to be reviewed, accepted, and merged into `master`. Once it's merged, I can adjust the PRs to be against `master`.

I probably should've just not mentioned those at all and just waited for this PR to pass.

**Comment by crstauf** [comment]
Discussed with @JPry in Slack... I'll get the `add` and `version` PRs closed and open new ones against `prospress/action-scheduler:master`.

**Comment by crstauf** [comment]
=== Picking up convo from https://github.com/Prospress/action-scheduler/pull/279#discussion_r271714990 here ===

@rrennick Which would be preferable?

A) implement a specific clause to the autoloader
B) move the `require` into main plugin file

**Comment by crstauf** [comment]
Opened #280 to continue this.

**Pull request merged**